gmsh_utils/gmsh_IO.py

```diff
@@ -400,7 +478,7 @@ def generate_extract_mesh(self, dims: int, mesh_name: str, mesh_output_dir: str,
 
```

Comment:
it holds. Overal, gmsh is a bit dangerous, because there are situations where gmsh doesnt close while python does. It is very important to use finalize_gmsh when needed